tools/install-powershell.ps1

```diff
@@ -79,6 +79,68 @@ Function Remove-Destination([string] $Destination) {
     }
 }
 
+Function IsPathInRegistryPath([string] $Path) {
+    #Remove ending DirectorySeparatorChar for comparison purposes
+    $Path = [System.Environment]::ExpandEnvironmentVariables([regex]::Replace($Path, '\\;$|\\$', ''));
+    $InstalledPaths = @()
+    #Possible gotcha: don't use Get-ItemPropertyValue (see https://github.com/PowerShell/PowerShell/issues/5906)
+    #Possible gotcha: coerce the (possibly null) property to string before calling ExpandEnvironmentVariables
+    [string] $HKCURegKeyPath = Get-ItemProperty -Path 'HKCU:\Environment' -Name 'PATH' -ErrorAction SilentlyContinue
+    $InstalledPaths += @([System.Environment]::ExpandEnvironmentVariables(($HKCURegKeyPath)).Split(';'))
```

Comment:
Don't expand paths

Reply:
You should not expand any part of the path you did not add.

Reply:
> You should not expand any part of the path you did not add.

This function checks if the path is already in the users path.  **It need to expand everything.**  

If $Destination is "%APPDATA%\1" and the PATH already has "%USERPROFILE%\Appdata\Roaming\1" then I don't want to add $Destination to the Path.... it's already in there.

In any event, the proposed change ([System.Environment]::GetEnvironmentVariable('PATH','User')) expands all variables so this change was commited in https://github.com/PowerShell/PowerShell/pull/8081/commits/2fe05aa5d07c220ac077c1fc52ef4231aad66286